### PR TITLE
Add datatype tests for empty

### PR DESCRIPTION
**Pull request by rafonsecad**
Adding `empty` tests on records as @NoahTheDuke suggested

### CODE REVIEW

File: test/clojure/core_test/empty.cljc [hunk 1]
```diff
@@ -24,4 +24,11 @@
            nil r/max-int
            nil r/min-int
            #?@(:cljs [nil (js/Date)]
-               :clj  [nil (new Object)])))))
+               :clj  [nil (new Object)])))
+
+    (testing "datatypes"
+      (defrecord Record [field])
+      (deftype Type [field])
```
Comment (jeaye) [inline test/clojure/core_test/empty.cljc:31]: Each test should try to use as few other `clojure.core` vars as possible, since those vars may not exist on whichever dialect is running test. On top of that, I'm not sure if every major Clojure dialect implements `defrecord` and `deftype`.

I think it's good thinking to add these, but I'm currently thinking this is doing too much for the `empty` test. Anyone have thoughts on that?

@rafonsecad @NoahTheDuke

Comment (rafonsecad) [reply test/clojure/core_test/empty.cljc:31]: yeah, that could be too much for `empty` test. In the other hand, we could add `when-var-exists` for `defrecord` and `deftype` to avoid the test from failing if those datatypes are not yet implemented

Comment (jeaye) [reply test/clojure/core_test/empty.cljc:31]: Good thoughts, folks. Yeah, let's use a `when-var-exists` for each of these vars. Mainly, I want these tests to also serve as an onboarding ramp for new dialects (like jank), so ideally each test has the smallest footprint it can.

Comment (rafonsecad) [reply test/clojure/core_test/empty.cljc]: @jeaye ok I just updated the pull request. Let me know if we need another change

Comment (rafonsecad) [reply test/clojure/core_test/empty.cljc]: @jeaye  yeah that makes sense. check again please

**Review by jeaye — APPROVED** [review]
Looks good to me. @NoahTheDuke?

**Comment by jeaye** [comment]
Thanks, @rafonsecad!
